addons/l10n_es_edi_tbai/models/l10n_es_edi_tbai_document.py

```diff
@@ -383,7 +383,7 @@ def _get_recipient_values(self, partner):
         if not partner._l10n_es_is_foreign() and partner.vat:
             recipient_values['nif'] = partner.vat[2:] if partner.vat.startswith('ES') else partner.vat
 
-        elif partner.country_id in self.env.ref('base.europe').country_ids:
+        elif partner.country_id and 'EU' in partner.country_id.country_group_codes:
```

Comment:
Wrong.
You will also match the Eurasian Economic Union (code=EEU).
All this substring search are wrong.

I propose to add `belongs_to()` method on `res.country`.

```python
def belongs_to(self, country_group_code):
    if not self:
        return False
    self.ensure_one()
    return country_group_code in self.country_group_ids.mapped('code')
```

```suggestion
        elif partner.country_id.belongs_to('EU'):
```

Reply:
@KangOl Indeed you are correct. But the method wouldn't be useful in case of the views (where we hide section as per group_codes not sure if the method would be accessible from view). Wondering what else we can do....

Well I found one hack not sure if it's good enough to use it here
https://github.com/odoo/odoo/blob/c434a38ab676ed8d65d9b000898fab36dd00f520/addons/account_check_printing/views/account_journal_views.xml#L30

```xml
        invisble="',EU,' not in company_country_group_codes"
```
Let me know if you have better way to do it

Reply:
It will work but it's fragile.

Reply:
@hamo-odoo either we need fixed length or add ,     Indeed, I did not think about that the regular fiscal_country_codes have a fixed length.

Reply:
"'EU'" not in country_codes if you put everything between ' and in theory we need to escape...